python/dicom_archive.py

```diff
@@ -0,0 +1,325 @@
+    )
+
+# Connect to database (if needed)
+
+db = None
+if profile is not None:
+    db = Database(load_config_file(profile).mysql, False)
+    db.connect()
+
+# Check paths
+
+while source.endswith('/'):
+    source = source[:-1]
+
+base_name = os.path.basename(source)
+
+tar_path     = f'{target}/{base_name}.tar'
+zip_path     = f'{target}/{base_name}.tar.gz'
+summary_path = f'{target}/{base_name}.meta'
+log_path     = f'{target}/{base_name}.log'
+
+check_create_file(tar_path)
+check_create_file(zip_path)
+check_create_file(summary_path)
+check_create_file(log_path)
+
+print('Extracting DICOM information (may take a long time)')
+
+summary = lib.dicom.summary_make.make(source, verbose)
+
+if db is not None:
+    print('Checking database presence')
+
+    archive = lib.dicom.dicom_database.get_archive_with_study_uid(db, summary.info.study_uid)
+
+    if db_insert and archive is not None:
+        print_error_exit(
+            (
+                f'Study \'{summary.info.study_uid}\' is already inserted in the database\n'
+                'Previous archiving log:\n'
+                f'{archive[1]}'
+            ),
+            lib.exitcode.INSERT_FAILURE,
+        )
+
+    if db_update and archive is None:
+        print_error_exit(
+            f'No study \'{summary.info.study_uid}\' found in the database',
+            lib.exitcode.UPDATE_FAILURE,
+        )
+else:
+    # Placeholder for type checker
+    archive = None
+
+print('Copying into DICOM tar')
+
+with tarfile.open(tar_path, 'w') as tar:
+    for file in os.listdir(source):
+        tar.add(source + '/' + file)
+
+print('Calculating DICOM tar MD5 sum')
+
+tarball_md5_sum = lib.dicom.text.make_hash(tar_path, True)
+
+print('Zipping DICOM tar (may take a long time)')
+
+with open(tar_path, 'rb') as tar:
+    with gzip.open(zip_path, 'wb') as zip:
+        shutil.copyfileobj(tar, zip)
+
+print('Calculating DICOM zip MD5 sum')
+
+zipball_md5_sum = lib.dicom.text.make_hash(zip_path, True)
+
+print('Getting DICOM scan date')
+
+if not today and summary.info.scan_date is None:
+    print_warning((
+        'No scan date found for this DICOM archive, '
+        'consider using argument \'--today\' to use today\'s date instead.'
+    ))
+
+scan_date = date.today() if today else summary.info.scan_date
+
+if year:
+    if not scan_date:
+        print_error_exit(
+            'Cannot use year directory with no date found for this DICOM archive.',
+            lib.exitcode.CREATE_DIR_FAILURE,
+        )
+
+    scan_date = cast(date, scan_date)
+
+    dir_path = f'{target}/{scan_date.year}'
+    if not os.path.exists(dir_path):
+        print(f'Creating directory \'{dir_path}\'')
+        os.mkdir(dir_path)
+    elif not os.path.isdir(dir_path) or not os.access(dir_path, os.W_OK):
+        print_error_exit(
+            f'Path \'{dir_path}\' exists but is not a writable directory.',
+            lib.exitcode.CREATE_DIR_FAILURE,
+        )
+else:
+    dir_path = target
```

Comment:
I am not super confortable using the date the pipeline is run to organize the DICOM.tar.gz files. What happens if the data gets run again through the pipeline on a different year? You would end up with duplicated files on the file system and that might cause confusions. Also, DICOMs are organized by acquisition date year later on in the pipeline, so that might mess up later code.

@nicolasbrossard, @laemtl  thoughts in this?